### ISSUE TITLE
Support standard Docker Registry as mirror

### ISSUE THREAD

**Issue by abulford** [vmware/harbor#3035]
As far as I can tell, Harbor can only mirror to other Harbor instances, when I try to point Harbor at a normal Docker Registry it connects OK but when actually trying to push an image it fails because it's unable to create the project.

Given that Docker Registry by itself is more lightweight than a standard Harbor set up - not running any database or UI - it would be great if Harbor could detect that it's connecting to a plain Docker Registry and just attempt to push the image without setting up the project first.

Here are the logs I get when attempting to mirror images from Harbor to a standard Docker Registry:

jobservice.log:
```
Aug  7 10:44:45 172.21.0.1 jobservice[1749]: 2017-08-07T09:44:45Z [WARNING] Job id: 2, the statemachin will enter error state due to error: failed to create project library on http://real-registry:5000 with user : 404 404 page not found
```

Docker Registry log:
```
time="2017-08-07T09:44:45Z" level=info msg="response completed" go.version=go1.7.3 http.request.host="real-registry:5000" http.request.id=ff6affab-2ba3-416f-a97d-2bc933287951 http.request.method=GET http.request.remoteaddr="172.21.0.7:41526" http.request.uri="/v2/" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=1.507536ms http.response.status=200 http.response.written=2 instance.id=3ef5fafb-5205-4d2e-92e2-2d5cbf498344 version=v2.6.0 
172.21.0.7 - - [07/Aug/2017:09:44:45 +0000] "GET /v2/ HTTP/1.1" 200 2 "" "Go-http-client/1.1"
172.21.0.7 - - [07/Aug/2017:09:44:45 +0000] "POST /api/projects/ HTTP/1.1" 404 19 "" "Go-http-client/1.1"
```

I've been trying this with Harbor v1.1.2.

**Comment by reasonerjt** [comment]
@abulford by "mirror" do you mean "replicate"?

**Comment by abulford** [comment]
@reasonerjt yes.  Sorry, I've no idea why I used the term 'mirror', which I understand is a different feature which allows local caching from a remote registry.  I can edit the original ticket if you think it will reduce confusion for future readers?

**Comment by draeron** [comment]
i've been trying to do exactly this (setup a vanilla docker registry as a target for replication).  I did not realised it wasn't supported as the documentation doesn't mention you need another instance of harbor. Might be a good idea to update the doc on this topic.

harbor version : 1.4.0
docker-registry : 2.6.2

both deployed on k8s via helm chart. i configure the second registry to use harbor for token authentication.

**Comment by clouderati** [comment]
We've not tested this particular setup. @draeron and @abulford, can you please share your use case?

**Comment by draeron** [comment]
we have multiple k8s clusters, one of them is hosting the Harbor service. We'd like to setup a vanilla docker registry in other clusters where the k8s nodes can pull the image from inside the cluster. The clusters runs in their own VPC so communication between them would be restricted for security purpose.

Ideally, once you push a new image on Harbor, it would get replicated on the "satellite" clusters and these wouldn't need to pull the images from the harbor.

**Comment by abulford** [comment]
@clouderati sorry for the delayed response.

We have a single instance of Harbor which is where Docker images get pushed to by the build system, we have several data centres across the world which each run a standard Docker Registry.  Docker hosts in these data centres pull from their local Docker Registry, rather than each of them pulling from Harbor across the WAN.

We want to use the standard Docker Registry as a mirror because we have no need for the UI and user management aspects that Harbor brings in these data centres, and this functionality requires several additional containers which consume resources, require effort to configure, update, etc.
